lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/persistence/DBOMembershipInvitation.java

```diff
@@ -1,276 +1,300 @@
+				if (rs.wasNull()) inviteeId=null;
+				dbo.setInviteeId(inviteeId);
+				String inviteeEmail = rs.getString(COL_MEMBERSHIP_INVITATION_INVITEE_EMAIL);
+				if (rs.wasNull()) inviteeEmail=null;
+				dbo.setInviteeEmail(inviteeEmail);
+				java.sql.Blob blob = rs.getBlob(COL_MEMBERSHIP_INVITATION_PROPERTIES);
+				if(blob != null){
+					dbo.setProperties(blob.getBytes(1, (int) blob.length()));
+				}
+				return dbo;
+			}
+
+			@Override
+			public String getTableName() {
+				return TABLE_MEMBERSHIP_INVITATION;
+			}
+
+			@Override
+			public String getDDLFileName() {
+				return DDL_FILE_MEMBERSHIP_INVITATION;
+			}
+
+			@Override
+			public FieldColumn[] getFieldColumns() {
+				return FIELDS;
+			}
+
+			@Override
+			public Class<? extends DBOMembershipInvitation> getDBOClass() {
+				return DBOMembershipInvitation.class;
+			}
+			
+		};
+	}
+
+	public Long getId() {
+		return id;
+	}
+
+	public void setId(Long id) {
+		this.id = id;
+	}
+
+	public String getEtag() {
+		return etag;
+	}
+
+	public void setEtag(String etag) {
+		this.etag = etag;
+	}
+
+	public Long getCreatedOn() {
+		return createdOn;
+	}
+
+	public void setCreatedOn(Long createdOn) {
+		this.createdOn = createdOn;
+	}
+
+	public byte[] getProperties() {
+		return properties;
+	}
+
+	public void setProperties(byte[] properties) {
+		this.properties = properties;
+	}
+
+	public Long getTeamId() {
+		return teamId;
+	}
+
+	public void setTeamId(Long teamId) {
+		this.teamId = teamId;
+	}
+
+	public Long getExpiresOn() {
+		return expiresOn;
+	}
+
+	public void setExpiresOn(Long expiresOn) {
+		this.expiresOn = expiresOn;
+	}
+
+	public Long getInviteeId() {
+		return inviteeId;
+	}
+
+	public void setInviteeId(Long inviteeId) {
+		this.inviteeId = inviteeId;
+	}
+
+	public String getInviteeEmail() {
+		return inviteeEmail;
+	}
+
+	public void setInviteeEmail(String inviteeEmail) {
+		this.inviteeEmail = inviteeEmail;
+	}
+
+	@Override
+	public MigrationType getMigratableTableType() {
+		return MigrationType.MEMBERSHIP_INVITATION;
+	}
+
+	@Override
+	public MigratableTableTranslation<DBOMembershipInvitation, DBOMembershipInvitation> getTranslator() {
+		return new MigratableTableTranslation<DBOMembershipInvitation, DBOMembershipInvitation>(){
+
+			@Override
+			public DBOMembershipInvitation createDatabaseObjectFromBackup(DBOMembershipInvitation backup) {
+				if (backup.getEtag() == null) {
+					backup.setEtag(defaultEtag);
+				}
+				try {
+					String backupProperties = new String(unzip(backup.getProperties()));
+					if (backupProperties.contains("MembershipInvtnSubmission")) {
+						// This backup object has a MembershipInvtnSubmission in its properties field
+						// Set its properties to be a MembershipInvitation instead
+						String correctedProperties = backupProperties.replace("MembershipInvtnSubmission", "MembershipInvitation");
+						backup.setProperties(zip(correctedProperties.getBytes()));
```

Comment:
We need to make sure that the result of the transformation is valid.  There are two approaches.  The most conservative approach is:
1) deserialize the properties into a MembershipInvtnSubmission object, the class ONLY being defined in the migrator and used no where else in Synapse.  (Copy the auto-generated POJO here.)
2) Copy fields from the MembershipInvtnSubmission object into the MembershipInvitation object;
3) Serialize the new MembershipInvitation object into the properties field of the DBO.

Another approach is to validate the correctedProperties: call MembershipInvitationUtils.deserialize(correctedProperties.getBytes());

A third approach is to use the name 'aliasing' feature of XStream to directly deserialize the legacy object into a MembershipInvitation object.